source/guides/clear/swupd-3rd-party.rst

```diff
@@ -0,0 +1,273 @@
+
+Prerequisite
+*************
+
+* Familiarity with :command:`mixer`
+* Familiarity with :command:`swupd`
+* You must be running |CL| version 32570 or higher
+
+.. include:: ./mixer.rst
+   :start-after: set-up-nginx-web-server-start:
+   :end-before: set-up-nginx-web-server-end:
+
+Create directory to hold 3rd-party app
+**************************************
+
+#. Create a top-level directory to hold all apps.
+	
+   .. code-block:: bash
+      
+      mkdir ~/my-3rd-party-apps && pushd $_
+
+#. Create a directory for each app and put the content of your software in it.  
+
+   In this example, the `helloclear.sh` app simply prints
+   "Hello Clear!" when invoked.  
+
+   .. code-block:: bash
+
+      # make helloclear directory
+      mkdir -p helloclear/usr/bin && pushd $_
+
+      # create helloclear.sh script
+      cat > helloclear.sh << EOF
+      #!/bin/bash
+      echo "Hello Clear!"
+      EOF
+
+      # make script executable
+      chmod +x helloclear.sh
+
+      popd
+
+   .. note:: 
+      
+      * You can put whatever you want in your app's directory. All of the content 
+        within each directory will get copied onto the client system under 
+        :file:`/opt/3rd-party/<app-directory>`.
+
+      * To use a 3rd-party RPM, it is recommended to extract the content of the RPM 
+        into a directory.  Use :command:`rpm2cpio <RPM>| sudo cpio -idv`.
+       
+Create bundle of 3rd-party app with mixer
+*****************************************
+
+Next, use :command:`mixer` to create a bundle for each of the apps from the previous 
+section.  
+
+#. Install the mixer tool.
+   
+   .. code-block:: bash
+
+      sudo swupd bundle-add mixer
+
+#. Create a mixer workspace.
+   
+   .. code-block:: bash
+
+      mkdir ~/mixer && cd $_
+
+#. Initialize a mix without any default bundles.
+   
+   .. code-block:: bash
+
+      mixer init --no-default-bundles
+
+#. Configure :file:`builder.conf` to set the default bundle, CONTENTURL, and VERSIONURL.  
+   For the "URL"s in this example, it will be IP address of the web server that was
+   set up earlier.
+   Substitute <IP-address-of-web-server> with the IP address of your host.  
+
+   .. code-block:: bash
+      
+      mixer config set Swupd.BUNDLE "os-core"
+      mixer config set Swupd.CONTENTURL "http://<IP-address-of-web-server>"
+      mixer config set Swupd.VERSIONURL "http://<IP-address-of-web-server>"
+
+#. Create an empty local `os-core` bundle.  :command:`swupd` client expects the 
+   `os-core` bundle to exist in a mix even if it’s empty.
+
+   .. code-block:: bash
+      
+      mixer bundle create os-core --local
+
+#. Using the `helloclear` app as an example, create the `helloclear` bundle  
+   and use the `content()` directive with the path to the `helloclear` directory in
+   the bundle definition.
+
+   Refer to `bundle definition`_ for addition information on how to define a bundle.
+
+   .. code-block:: bash
+      
+      mixer bundle create helloclear
```

Comment:
I would suggest including the local flag over here too. In case they add a bundle called `helloclear`  upstream in future.

> mixer bundle create helloclear --local

Reply:
done.